### PR TITLE
fmtoy: Update, init split-off projects

### PR DESCRIPTION
**Pull request by OPNA2608**
## Description of changes

Closes #227539.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/by-name/vg/vgm2x/package.nix [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  '';
+
+  passthru = {
+    updateScript = unstableGitUpdater { };
+  };
+
+  meta = with lib; {
+    description = "VGM file extraction tools";
+    homepage = "https://github.com/vampirefrog/vgm2x";
+    # Not clarified of GPL3-Only or GPL3-Plus, assuming the former
```
Comment (SuperSandro2000) [inline pkgs/by-name/vg/vgm2x/package.nix:52]:
```suggestion
```
This is standard procedure

File: pkgs/by-name/li/libfmvoice/package.nix [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  '';
+
+  passthru = {
+    updateScript = unstableGitUpdater { };
+  };
+
+  meta = with lib; {
+    description = "C library for loading, saving and converting FM sound chip voice files in various formats";
+    homepage = "https://github.com/vampirefrog/libfmvoice";
+    # Not clarified if GPL3-Only or GPL3-Plus, assuming the former
```
Comment (SuperSandro2000) [inline pkgs/by-name/li/libfmvoice/package.nix:48]:
```suggestion
```
see below

File: pkgs/by-name/li/libfmvoice/package.nix [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  buildFlags = [
+    "CC=${stdenv.cc.targetPrefix}cc"
+  ];
+
+  installPhase = ''
+    runHook preInstall
+
+    for prog in $(grep 'PROGS=' Makefile | cut -d'=' -f2); do
+      install -Dm755 $prog $out/bin/$prog
+    done
```
Comment (SuperSandro2000) [inline pkgs/by-name/li/libfmvoice/package.nix:36]: Why do people have no standard install target?! sigh

File: pkgs/by-name/fm/fmtoy/package.nix [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+  passthru = {
+    updateScript = unstableGitUpdater { };
+  };
+
+  meta = with lib; {
+    description = "FM synthesiser based on emulated Yamaha YM chips (OPL, OPM and OPN series)";
+    homepage = "https://github.com/vampirefrog/fmtoy";
+    # Unclear if gpl3Only or gpl3Plus, assuming the former
+    # https://github.com/vampirefrog/fmtoy/issues/1
```
Comment (SuperSandro2000) [inline pkgs/by-name/fm/fmtoy/package.nix:66]:
```suggestion
```
as below, if there is no mention of later it is only

File: pkgs/by-name/li/libfmvoice/package.nix (hunk 3 above)
Comment (OPNA2608) [reply pkgs/by-name/li/libfmvoice/package.nix:36]: Different priorities I suppose. I feel like complaining about it here isn't a nice practice either…

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by OPNA2608** [comment]
Bumped `libfmvoice` to fix Darwin builds.

**Comment by pbsds** [comment]
Result of `nixpkgs-review pr 246771` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>fmtoy</li>
    <li>libfmvoice</li>
    <li>vgm2x</li>
  </ul>
</details>



- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] package builds on x86_64-linux
- [x] executables tested on x86_64-linux
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.platforms` is set
- [x] `meta.maintainers` is set
- [x] build time only dependencies are declared in `nativeBuildInputs`
- [x] source is fetched using the appropriate function
- [x] the list of `phases` is not overridden
- [x] when a phase (like `installPhase`) is overridden it starts with `runHook preInstall` and ends with `runHook postInstall`.
- [x] patches have a comment describing either the upstream URL or a reason why the patch wasn't upstreamed
- [x] patches that are remotely available are fetched rather than vendored

LGTM, thanks!

**Pull request merged**